File: 14-Facade/src/go/projector.go
package main

import "fmt"

// Subsystem type — the video projector.
type Projector struct{}

func (p Projector) On()              { fmt.Println("Projector on") }
func (p Projector) Off()             { fmt.Println("Projector off") }
func (p Projector) WideScreenMode()  { fmt.Println("Projector in widescreen mode (16:9)") }

File: 14-Facade/src/rust/projector.rs
// Subsystem — the video projector.
pub struct Projector;

impl Projector {
    pub fn on(&self) { println!("Projector on"); }
    pub fn off(&self) { println!("Projector off"); }
    pub fn wide_screen_mode(&self) { println!("Projector in widescreen mode (16:9)"); }
}
